eventkitd/writer.go

```diff
@@ -56,8 +63,17 @@ func (w *Writer) Append(path string, data []byte) error {
 			w.mtx.Unlock()
 			return err
 		}
+
+		rcw, err := resumablecompressed.NewWriter(fh, zlib.DefaultCompression)
+		if err != nil {
+			fh.Close()
```

Comment:
This close can be done without holding the mutex.

Reply:
oh good point

Reply:
Also, shouldn't we try to delete the created file as well?

Reply:
i don't think it matters. it should be safe to append to an existing file

Reply:
Not unless the file is locked for writing.

Reply:
Yeah, AFAIR, unless you do some extra work (FILE_SHARE_WRITE) then Windows doesn't let you open the same file for writing. But, I'll verify it.

Reply:
Ok, nevermind... It does seem that it does work on Windows as well. So, I'll mark it as resolved.